primitives/src/lib.rs

```diff
@@ -28,19 +28,29 @@ extern crate std;
 extern crate actual_serde as serde;
 
 #[cfg(feature = "alloc")]
-pub mod locktime;
+pub mod locktime; // Maintains units module structure and re-exports everything.
 pub mod opcodes;
 pub mod sequence;
 
-#[rustfmt::skip]                // Keep public re-exports separate.
 #[doc(inline)]
 #[cfg(feature = "alloc")]
-pub use self::{
-    locktime::{absolute, relative},
-};
+pub use self::locktime::{absolute, relative};
 #[doc(inline)]
 pub use self::sequence::Sequence;
 
+// Mirror the module structure and re-exports of `units`.
+#[rustfmt::skip]                // Keep public unit re-exports separate.
+#[doc(inline)]
+pub use units::amount::{self, Amount, SignedAmount};
+#[cfg(feature = "alloc")]
+#[doc(inline)]
+pub use units::{
+    block::{self, BlockHeight, BlockInterval},
+    fee_rate::{self, FeeRate},
+    parse,
+    weight::{self, Weight},
+};
```

Comment:
IMO these should be written to reference them from crate root to demonstrate that the structure is indeed the same. Though maybe we can just `pub use units::*;` in the root and don't worry about anything else.

Reply:
Ok, so we have nailed down what we want but not how, can we merge this as is (it does the what) and leave the how for #3086. This PR is blocking any other `primitives` work and has been open for over 5 days already.